hollow/src/main/java/com/netflix/hollow/api/consumer/incubating/HollowConsumerV2.java

```diff
@@ -0,0 +1,515 @@
+                builder.metrics,
+                builder.refreshLock
+        );
+        this.updaterV2 = builder.updater;
+    }
+
+    public CompletableFuture<HollowConsumerV2<A>> initialLoad() {
+        return getInitialLoad().thenApply(l -> this);
+    }
+
+    @Override
+    public A getAPI() {
+        return updaterV2.getAPI();
+    }
+
+    /**
+     * Equivalent to calling {@link #getAPI()} and casting to the specified API.
+     *
+     * @param apiClass the class of the API
+     * @param <T> the type of the API
+     * @return the API which wraps the underlying dataset
+     *
+     * @deprecated use type parameter on consumer, e.g. {@code HollowConsumerV2<T>} then call {@link #getAPI()}
+     */
+    @Override
+    @Deprecated
+    public <T extends HollowAPI> T getAPI(Class<T> apiClass) {
+        return apiClass.cast(updater.getAPI());
+    }
+
+    public static class Builder<A extends HollowAPI> {
+        final Feature<BlobRetriever> blobRetriever;
+        final Feature<Path> localBlobStore;
+        final Feature<AnnouncementWatcher> announcementWatcher;
+        final Feature<Set<RefreshListener>> refreshListeners;
+        final Feature<Class<A>> apiClass;
+        final Feature<String[]> cachedTypes;
+        final Feature<HollowFilterConfig> filterConfig;
+        final Feature<DoubleSnapshotConfig> doubleSnapshotConfig;
+        final Feature<ObjectLongevityConfig> objectLongevityConfig;
+        final Feature<ObjectLongevityDetector> objectLongevityDetector;
+        final Feature<Executor> refreshExecutor;
+        final Feature<HollowMetricsCollector<HollowConsumerMetrics>> metricsCollector;
+
+        HollowAPIFactory apiFactory;
+        HollowConsumerMetrics metrics;
+        ReadWriteLock refreshLock;
+        HollowClientUpdaterV2<A> updater;
+        HollowObjectHashCodeFinder hashCodeFinder;
+        List<RefreshListener> refreshListenerList;
+
+        protected Builder() {
+            blobRetriever           = feature();
+            localBlobStore          = feature();
+            announcementWatcher     = feature();
+            refreshListeners        = feature();
+            apiClass                = feature();
+            cachedTypes             = feature();
+            filterConfig            = feature();
+            doubleSnapshotConfig    = feature();
+            objectLongevityConfig   = feature();
+            objectLongevityDetector = feature();
+            refreshExecutor         = feature();
+            metricsCollector        = feature();
+        }
+
+        protected <T> Feature<T> feature() {
+            return new Feature<>();
+        }
+
+        public synchronized Builder<A> withBlobRetriever(BlobRetriever blobRetriever) {
+            requireNonNull(blobRetriever, "blobRetriever cannot be null; use blobRetrieverFeature().disable()");
+            return this.blobRetriever.set(blobRetriever);
+        }
+
+        public Feature<BlobRetriever> blobRetrieverFeature() {
```

Comment:
I want every (or most) features to have a `withFoo(...)` and `noFoo()` pair of methods. I also want to avoid using `withFoo(null)` as the mechanism for disabling a feature; that way a caller with a coding error doesn't silently disable a feature in their builder.

I realized along the way that I really want 3 states: unset/default, user customized, and disabled. I got partway through making every feature have a triple of `withFoo(...)`, `resetFoo()`, and `noFoo()` before I decided to try `fooFeature().reset()` or `fooFeature().disable()`. Maybe for the final public API the `with|reset|disable` is simpler and better; for now it's one less boilerplate method I have to write as I'm experimenting.